File: RedSheet.DbEntities/Models/vNegotionalityMember.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vNegotionalityMembers", Schema = "dbo")]
	[Serializable]
	public partial class vNegotionalityMember  
	{
 
	[Range(1, int.MaxValue)]
	public int AgreeableId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int AssertivenessId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ConflictStyleId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ConsciousnessId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int EmotionalCompetenceId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int OpenMindedId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int OurTeamMemberBehaviourId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int OurTeamMemberId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int OurTeamMemberRequireId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int OutingId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectNegotionalityId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int RequireAgreeableId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int RequireAssertivenessId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int RequireConflictStyleId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int RequireConsciousnessId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int RequireEmotionalCompetenceId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int RequireOpenMindedId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int RequireOutingId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TeamRoleId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string TeamRoleName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int UserId { get; set; } 
 
	[Required] 
	[MaxLength(201)]
	public string UserName { get; set; } 

	public vNegotionalityMember()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-negotionality-member.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vNegotionalityMember {
    constructor(vNegotionalityMember?: vNegotionalityMember )  {
        let properties = [ "agreeableId", "assertivenessId", "conflictStyleId", "consciousnessId", "emotionalCompetenceId", "openMindedId", "ourTeamMemberBehaviourId", "ourTeamMemberId", "ourTeamMemberRequireId", "outingId", "projectNegotionalityId", "requireAgreeableId", "requireAssertivenessId", "requireConflictStyleId", "requireConsciousnessId", "requireEmotionalCompetenceId", "requireOpenMindedId", "requireOutingId", "teamRoleId", "teamRoleName", "userId", "userName",];
        for (let property of properties)
            if (vNegotionalityMember && vNegotionalityMember[property])
                this[property] = vNegotionalityMember[property];
    }
 
    @range(1,2147483647)
	agreeableId : number =   undefined;
 
    @range(1,2147483647)
	assertivenessId : number =   undefined;
 
    @range(1,2147483647)
	conflictStyleId : number =   undefined;
 
    @range(1,2147483647)
	consciousnessId : number =   undefined;
 
    @range(1,2147483647)
	emotionalCompetenceId : number =   undefined;
 
    @range(1,2147483647)
	openMindedId : number =   undefined;
 
	ourTeamMemberBehaviourId : number =   0 ;
 
    @range(1,2147483647)
	ourTeamMemberId : number =   undefined;
 
    @range(1,2147483647)
	ourTeamMemberRequireId : number =   undefined;
 
    @range(1,2147483647)
	outingId : number =   undefined;
 
    @range(1,2147483647)
	projectNegotionalityId : number =   undefined;
 
    @range(1,2147483647)
	requireAgreeableId : number =   undefined;
 
    @range(1,2147483647)
	requireAssertivenessId : number =   undefined;
 
    @range(1,2147483647)
	requireConflictStyleId : number =   undefined;
 
    @range(1,2147483647)
	requireConsciousnessId : number =   undefined;
 
    @range(1,2147483647)
	requireEmotionalCompetenceId : number =   undefined;
 
    @range(1,2147483647)
	requireOpenMindedId : number =   undefined;
 
    @range(1,2147483647)
	requireOutingId : number =   undefined;
 
    @range(1,2147483647)
	teamRoleId : number =   undefined;
 
    @required()
    @maxLength(200)
	teamRoleName : string =   undefined;
 
    @range(1,2147483647)
	userId : number =   undefined;
 
    @required()
    @maxLength(201)
	userName : string =   undefined;


}
